docs/docs/01-ibc/10-roadmap.md

```diff
@@ -7,35 +7,166 @@ slug: /ibc/roadmap
 
-We will refactor the 05-port router to enable atomic sending of multiple packets belonging to different applications.
+```proto
+message MsgSendPacket {
+  string                    source_port       = 1;
+  string                    source_channel    = 2;
+  ibc.core.client.v1.Height timeout_height    = 3 [(gogoproto.nullable) = false];
+  uint64                    timeout_timestamp = 4;
+  bytes                     data              = 5;
+  string                    signer            = 6;
+}
+```
+
+For more details, please check [#7012](https://github.com/cosmos/ibc-go/issues/7012).
+
+### Multipacket execution
+
+Building on top of the 05-port refactor, we will implement support for multiple packet data's to be delivered and executed within a single packet. This will allow applications wired together in a list of ordered applications to provide their own packet data. By introducing this feature, we are then able to specify in the packet the encoding and the version of each application's packet data, which will enable us to remove in the future channel handshakes and channel upgradability (as each packet data will indicate which application it is bound for, what version of the application it wishes to interact with and how it can be decoded).
+
+Currently the `Packet` proto message look like the following:
+
+```proto
+message Packet {
+  option (gogoproto.goproto_getters) = false;
+  uint64 sequence = 1;
+  string source_port = 2;
+  string source_channel = 3;
+  string destination_port = 4;
+  string destination_channel = 5;
+  bytes data = 6;
+  ibc.core.client.v1.Height timeout_height = 7 [(gogoproto.nullable) = false];
+  uint64 timeout_timestamp = 8;
+}
+```
 
----
+Where the `data` field is the byte representation of the application packet data (e.g the bytes that result from serializing `FungibleTokenPacketData` in the case of transfer). This packet structure supports only a single application packet data. With multipacket support we will give each application in a ordered lists of applications the possibility to have their own packet data. Therefore, we will introduce a new packet structure that might look like this:
+
+```proto
+message PacketV2 {
+  int64                     sequence            = 1;
+  string                    source_port         = 2;
+  string                    source_channel      = 3;
+  string                    destination_port    = 4;
+  string                    destination_channel = 5;
+  repeated PacketData       data                = 6; // primary modification
+  ibc.core.client.v1.Height timeout_height      = 7 [(gogoproto.nullable) = false];
+  uint64                    timeout_timestamp   = 8;
+}
+
+message PacketData {
+  string port_id  = 1; // or app_name
+  string type     = 2; // or version
+  string encoding = 3;
+  bytes  value    = 4; // the serialized bytes of the application's packet data
+}
```

Comment:
I took the liberty to put all this together in the same message (I know in the issue it's separate in two different structures, but thought that for clarity it would be possible to put it all together).